curator-framework/src/test/java/org/apache/curator/framework/imps/TestWatchesBuilder.java

```diff
@@ -611,6 +614,56 @@ public void pathAddedForGuaranteedOperation(
         }
     }
 
+    @Test(groups = CuratorTestBase.zk36Group)
+    public void testPersistentRecursiveWatch() throws Exception
+    {
+        try ( CuratorFramework client = CuratorFrameworkFactory.newClient(server.getConnectString(), new RetryOneTime(1)) )
+        {
+            client.start();
+            client.blockUntilConnected();
+
+            CountDownLatch latch = new CountDownLatch(5);
+            Watcher watcher = event -> latch.countDown();
+            client.watchers().add().withMode(AddWatchMode.PERSISTENT_RECURSIVE).usingWatcher(watcher).forPath("/test");
+
+            client.create().forPath("/test");

```

Comment:
Not interesting to test here also other recursive events like data changes and remove nodes?
Also, might be good to create and modify recursively "/not-the-test" and make sure nothing comes from there.
And remove watcher tests shouldn't be repeated for persistent/recursive too?

Reply:
I was thinking it isn't necessary to test functionality that's already tested in ZK itself. But, this is a good point. I'll add some more tests.